src/layman/layer/rest_layer_patch_test.py

```diff
@@ -0,0 +1,187 @@
+import pytest
+import importlib
+
+from layman import settings, app
+from layman.common import geoserver
+from layman.layer.prime_db_schema import table as prime_db_schema
+from layman.common.prime_db_schema import users
+
+from test import process, process_client as client_util
+
+
+liferay_mock = process.liferay_mock
+LIFERAY_PORT = process.LIFERAY_PORT
+
+ISS_URL_HEADER = client_util.ISS_URL_HEADER
+TOKEN_HEADER = client_util.TOKEN_HEADER
+
+AUTHN_INTROSPECTION_URL = process.AUTHN_INTROSPECTION_URL
+
+AUTHN_SETTINGS = process.AUTHN_SETTINGS
+
+
+def assert_gs_user_and_roles(username):
+    auth = settings.LAYMAN_GS_AUTH
+    gs_usernames = geoserver.get_usernames(auth)
+    assert username in gs_usernames
+    gs_user_roles = geoserver.get_user_roles(username, auth)
+    user_role = f"USER_{username.upper()}"
+    assert user_role in gs_user_roles
+    assert settings.LAYMAN_GS_ROLE in gs_user_roles
+
+
+def assert_gs_layer_data_security(username,
+                                  layername,
+                                  expected_roles):
+    auth = settings.LAYMAN_GS_AUTH
+    for right_type in ['read', 'write']:
+        gs_expected_roles = geoserver.layman_users_to_geoserver_roles(expected_roles[right_type])
+        gs_roles = geoserver.get_security_roles(f'{username}.{layername}.{right_type[0]}', auth)
+        assert gs_expected_roles == gs_roles\
+            or ('ROLE_ANONYMOUS' in gs_expected_roles and 'ROLE_ANONYMOUS' in gs_roles), right_type
```

Comment:
Přidat podmínku, že pak se liší jen v "username" na straně GS.

Reply:
Napadlo mě, že by možná bylo lepší na GS posílat stejná práva, jako vracíme v GET Layer access_rights. Tedy vždy explicitně s rolí ownera, pokud je to private workspace.